src/assets/assets.service.ts

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+import { BasePrismaClient } from '../prisma/types/base-prisma-client';
+import { CreateAssetOptions } from './interfaces/create-asset-options';
+import { Asset, Transaction } from '.prisma/client';
+
+@Injectable()
+export class AssetsService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async findByIdentifierOrThrow(
+    identifier: string,
+    prisma: BasePrismaClient,
```

Comment:
remove this, the prisma service is already present via injection

Reply:
I'm going to leave this for now, but remove the injection. The loader that I'm to add will pass this in for a database transaction. I will re-add the injection for the REST module later